### PR TITLE
Issue #303: Fix loose config issue with project dependencies handled by IvyDE

### LINKED ISSUE #303: Project dependencies managed by IvyDE do not get added to the loose config xml file
For an application which has its dependencies managed by the IvyDE plugin, if loose config mode is enabled (run applications directly from the workspace) the dependencies do not get added to the loose config xml file resulting in class not found exceptions.  The application deploys properly if loose config is disabled.

### PR DESCRIPTION
**Pull request by eharris369**
Fixes #303

### CODE REVIEW

**Comment by eharris369** [comment]
Ran all loose configuration tests against the fix (including Docker) and they all passed.

File: dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/looseconfig/LooseArchive.java [hunk 1]
```diff
@@ -336,27 +321,9 @@ else if (isMatchedComponentDependency(file, project.getLocation(), javaClasspath
...
-            }
-        }
-
-        return false;
-    }
-
     private static boolean canAddReferencedComponentClasspathDependencies(IModule module, IProject refProject) {
-        if ("jst.web".equals(module.getModuleType().getId())) {
+        String type = module.getModuleType().getId();
+        if ("jst.web".equals(type) || "jst.ear".equals(type)) {
```
Comment (elsony) [inline dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/looseconfig/LooseArchive.java]: Are we adding EAR support for shared library as part of this item? It looks like we were not allowing that before we make this change?

Comment (eharris369) [reply dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/looseconfig/LooseArchive.java]: It has always been supported for ear projects.  This code was added to fix a particular customer problem with a web project and I think whoever fixed that problem did not consider the ear case.  I was wondering if others should be added as well.  Looking at the plugin.xml for these properties it looks like they are shown for jst.web, jst.ear and jst.ejb.  So if anything I should be including jst.ejb as well:

   <page
   		id="com.ibm.ws.st.jee.ui.sharedLibraryWebProperties"
   		name="%sharedLibraryRefPropertyPage"
   		class="com.ibm.ws.st.jee.ui.internal.SharedLibPropertiesPage"
   		category="com.ibm.ws.st.ui.classpath">
   		<enabledWhen>
   		  <and>
   		   <adapt type="org.eclipse.core.resources.IProject">
   		   <test forcePluginActivation="false" property="com.ibm.ws.st.runtimeTarget"/>
   		   <or>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.web"/>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.ear"/>
            <test forcePluginActivation="true"
              property="org.eclipse.wst.common.project.facet.core.projectFacet"
                value="jst.ejb"/>
           </or>
           </adapt>
          </and>
   		</enabledWhen>
   	</page>
   </extension>

Comment (elsony) [reply dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/looseconfig/LooseArchive.java]: sounds good.

**Pull request merged**